railties/lib/rails/configuration.rb

```diff
@@ -64,9 +65,10 @@ def unshift(*args, &block)
       end
 
       def merge_into(other) #:nodoc:
-        @operations.each do |operation, args, block|
+        @operations.concat(@delete_operations).each do |operation, args, block|
```

Comment:
I think changing the operations array is fine since delete_operations are meant to be part of the initial array just that we're just appending the delete operations last to run them last.